ledger-service/cli-opts/src/main/scala/cliopts/Logging.scala

```diff
@@ -0,0 +1,38 @@
+// Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.cliopts
+
+import ch.qos.logback.classic.{Level => LogLevel}
+
+object Logging {
+
+  private val KnownLogLevels = Set("ERROR", "WARN", "INFO", "DEBUG", "TRACE")
+
+  implicit val scoptLogLevel: scopt.Read[LogLevel] = scopt.Read.reads { level =>
+    Either
+      .cond(
+        KnownLogLevels.contains(level.toUpperCase),
+        LogLevel.toLevel(level.toUpperCase),
+        s"Unrecognized logging level $level",
+      )
+      .getOrElse(throw new java.lang.IllegalArgumentException(s"Unknown logging level $level"))
+  }
+
+  /** Parse in the cli option for the logging level.
+    */
+  def loggingLevelParse[C](
+      parser: scopt.OptionParser[C]
+  )(logLevel: Setter[C, Option[LogLevel]]): Unit = {
+    import parser.opt
+
+    val _ =
+      opt[LogLevel]("log-level")
```

Comment:
```suggestion
    opt[LogLevel]("log-level")
```
While I’m usually a fan of `NonUnitStatement` warning and avoiding those here, `scopt` is one of those libraries where it’s just pointless to try since they are clearly not designed for this so I’d rather be consistent and ignore it everywhere instead of making this the odd one out.

Reply:
I was required to do that there if I don't want to suppress the warning.

Reply:
ok bit weird how it works but it seems like placing the `()` at the end of the method is enough for scalac not complaining about discarding non unit results.

Anyways, resolved.

Reply:
> placing the `()` at the end of the method is enough for scalac not complaining about discarding non unit results

This suppresses the `value-discard` warning built into scalac, but not `NonUnitStatements`, which is turned on for `http-json-*` but not `cli-opts`.